documentation/docs/20-core-concepts/10-routing.md

```diff
@@ -325,6 +325,31 @@ export async function POST({ request }) {
 
 > In general, [form actions](form-actions) are a better way to submit data from the browser to the server.
 
+> If a `GET` handler is exported, a `HEAD` request will return the `content-length` of the `GET` handler's response body.
+
+### Catch-all method
+
+Exporting the `all` handler will match any unhandled request methods.
+
+```js
+// @errors: 7031
+/// file: src/routes/api/add/+server.js
+import { json, text } from '@sveltejs/kit';
+
+export async function POST({ request }) {
+	const { a, b } = await request.json();
+	return json(a + b);
+}
+
+// This handler will respond to PUT, PATCH, DELETE, etc.
+/** @type {import('./$types').RequestHandler} */
+export async function all({ request }) {
+	return text(`I caught your ${request.method} request!`);
+}
+```
+
+> The `all` handler will also override the implicit `HEAD` handler that is added when a `GET` handler is exported.
```

Comment:
hmm. I'm not sure that's how we'd want this to behave?

Reply:
I think it's more consistent that way - `all` is a catch-all, no exceptions. Rich agrees: https://github.com/sveltejs/kit/issues/9164#issuecomment-1440835178

Reply:
I'm also not sure about the name `all` 😆  I agree with Rich that if you call it `all` you might want it to handle HEAD, but I'm not sure we should call it all. Seems like @dominikg also agrees it should not handle HEAD: https://github.com/sveltejs/kit/issues/9164#issuecomment-1443377519